base/src/main/java/org/mozilla/jss/provider/javax/crypto/JSSTrustManager.java

```diff
@@ -428,4 +438,25 @@ public X509Certificate[] getAcceptedIssuers() {
 
         return caCerts.toArray(new X509Certificate[caCerts.size()]);
     }
+
+    private void certChainRevokeVerify(X509Certificate[] certChain, String KeyUsage, ValidityStatus status) {
+        logger.debug("JSSTrustManager: certChainRevokeVerify()");
+        
+        //NOTE: if the usage is not correct the verification stops and the revocation is not verified
+        int leafUsage = KeyUsage.equals(CLIENT_AUTH_OID) ? CertificateUsage.SSLClient.getUsage() : CertificateUsage.SSLServer.getUsage();
```

Comment:
I wonder if we could pass a value that NSS will always consider valid so basically we're bypassing the key usage verification in NSS, then we implement our own key usage verification in PKI. We can deal with this later if it becomes a problem.

Reply:
I have done a lot of tests with many combination but I cannot get this passed. I have done also some investigation on NSS code but I have not found alternative APIs performing only the revocation check. We can implement this in JSS using lower NSS APIs but this can be evaluated in separate PR.